File: webapp/src/components/books.rs
use crate::prelude::*;

pub(crate) fn render_case_books(_contexts: Contexts) -> Html {
    html! {
        <CaseBooks />
    }
}

#[function_component(CaseBooks)]
pub(crate) fn case_books() -> Html {
    html! {
        <Paper class={format!("{} gap-1", CLASSES_SIDE_BY_SIDE)}>
            <Paper class="d-flex flex-column justify-center">
                <Quote color={Theme::Success} elevation={ELEVATION_STANDARD}>
                    {r#"
                    Get your copy of my book "CASE: Continous Agile Software Engineering" for more information about Continuous Agile Software Engineering and how to integrate it into your teams and workflows.

                    Note: These links use an affiliate tracker that allow us to earn an extra commission when you make a purchase."#}
                </Quote>
            </Paper>
            <Paper class="d-flex flex-row pa-2 gap-2 justify-center flex-wrap">
                <Paper elevation={ELEVATION_STANDARD} class="pa-2">
                    <Link target="_blank" href="https://amzn.to/48S30VU">
                        <Paper class="theme-secondary text-center pa-3">{"Kindle"}</Paper>
                    </Link>
                </Paper>
                <Paper elevation={ELEVATION_STANDARD} class="pa-2">
                    <Link target="_blank" href="https://amzn.to/3Sgpyu3">
                        <Paper class="theme-tertiary text-center pa-3">{"Paperback"}</Paper>
                    </Link>
                </Paper>
                <Paper elevation={ELEVATION_STANDARD} class="pa-2">
                    <Link target="_blank" href="https://amzn.to/3vue8db">
                        <Paper class="theme-primary text-center pa-3">{"Hardcover"}</Paper>
                    </Link>
                </Paper>
            </Paper>
        </Paper>
    }
}

File: webapp/wc/books.js
/* Display CASE books */
"use strict"
webui.define("app-books", {
    connected: (t) => {
        t.innerHTML = webui.applyAppDataToContent(`
<webui-side-by-side>

<webui-flex column justify="center">

<webui-quote theme="success">

Get your copy of the book "CASE: Continous Agile Software Engineering" for more information about Continuous Agile Software Engineering and how to integrate it into your teams and workflows.

Note: These links use an affiliate tracker that allow us to earn an extra commission when you make a purchase.

</webui-quote>

</webui-flex>

<webui-flex column align="center" class="ma-2">

<webui-flex gap="2" justify="center" wrap>

<webui-paper class="pa-2" elevation="10">

<webui-link target="_blank" href="https://a.co/d/0reKAjn" class="theme-secondary pa-3">

Kindle

</webui-link>

</webui-paper>

<webui-paper class="pa-2" elevation="10">

<webui-link target="_blank" href="https://a.co/d/9Kiyse8" class="theme-tertiary pa-3">

Paperback

</webui-link>

</webui-paper>

<webui-paper class="pa-2" elevation="10">

<webui-link target="_blank" href="https://a.co/d/3da8fQc" class="theme-primary pa-3">

Hardcover

</webui-link>

</webui-paper>

</webui-flex>

</webui-flex>

</webui-side-by-side>
`);
    }
});
